src/core/meshcop/tcat_agent.hpp

```diff
@@ -0,0 +1,382 @@
+                        uint8_t      aTlvType,
+                        uint32_t     aOffset,
+                        uint32_t     aLength,
+                        bool        &aHasResponse);
+    Error HandleCommissioning(ot::Message &aIncommingMessage,
+                              ot::Message &aOutgoingMessage,
+                              uint8_t      aTlvType,
+                              uint32_t     aOffset,
+                              uint32_t     aLength,
+                              bool        &aHasResponse);
+    Error HandleExtraction(ot::Message &aIncommingMessage,
+                           ot::Message &aOutgoingMessage,
+                           uint8_t      aTlvType,
+                           uint32_t     aOffset,
+                           uint32_t     aLength,
+                           bool        &aHasResponse);
+    Error HandleDecommissioning(ot::Message &aIncommingMessage,
+                                ot::Message &aOutgoingMessage,
+                                uint8_t      aTlvType,
+                                uint32_t     aOffset,
+                                uint32_t     aLength,
+                                bool        &aHasResponse);
+    Error HandleApplication(ot::Message &aIncommingMessage,
+                            ot::Message &aOutgoingMessage,
+                            uint8_t      aTlvType,
+                            uint32_t     aOffset,
+                            uint32_t     aLength,
+                            bool        &aHasResponse);
+    Error HandleCCM(ot::Message &aIncommingMessage,
+                    ot::Message &aOutgoingMessage,
+                    uint8_t      aTlvType,
+                    uint32_t     aOffset,
+                    uint32_t     aLength,
+                    bool        &aHasResponse);
+
+    Error HandleCommand(CommandType aCommand, ot::Message &aOutgoingMessage);
+    Error HandleActiveDataset(ot::Message &aIncommingMessage, uint32_t aOffset, uint32_t aLength);
+
+    static constexpr uint16_t kJoinerUdpPort = OPENTHREAD_CONFIG_JOINER_UDP_PORT;
+
+    JoinerPskd                       mJoinerPskd;
+    VendorInfo                      *mVendorInfo;
+    Callback<JoinCallback>           mJoinCallback;
+    Callback<AppDataReceiveCallback> mAppDataReceiveCallback;
+    CertificateAuthorizationField    mCommissionerAuthorizationField;
+    CertificateAuthorizationField    mDeviceAuthorizationField;
+    TcatMessageType                  mCurrentApplicationLayer;
+    char                             mCurrentServiceName[OT_TCAT_MAX_SERVICE_NAME_LENGTH + 1];
+    bool                             mEnabled : 1;
+    bool                             mConnected : 1;
+    bool                             mAlreadyCommissioned : 1;
+    bool                             mPskdVerified : 1;
+    bool                             mPskcVerified : 1;
+    bool                             mResponse : 1;
```

Comment:
mAlreadyCommissioned, mPskdVerified, mPskcVerified , mResponse will be needed for complete implementation of the TCAT specification. Some code pieces using this variables are still missing.

Reply:
Sounds good.

But if we plan to add the missing functionality (that would use these variable) in future PRs, it may be better not to include them in this PR and define them from the same PR that implements the functionality?

Reply:
Ok. Will save this for a future "feature completeness" PR.